### PR TITLE
feat: api v1 validator list

### PR DESCRIPTION
**Pull request by songwongtp**

### CODE REVIEW

File: src/lib/services/validator.ts [hunk 1]
```diff
@@ -74,3 +77,39 @@ export const resolveValIdentity = async (
       })
   );
 };
+
+const zValidatorsResponse = z
+  .object({
+    items: z.array(zValidatorData),
+    total: z.number().nonnegative(),
+    total_voting_power: z.string(),
```
Comment (evilpeach) [inline src/lib/services/validator.ts:85]: Let's create zBig and apply to `total_voting_power`.

File: src/lib/services/validator.ts [hunk 2]
```diff
@@ -74,3 +77,39 @@ export const resolveValIdentity = async (
...
+  })
+  .transform(({ items, total, total_voting_power }) => ({
```
Comment (evilpeach) [inline src/lib/services/validator.ts:87]: By using zBig, we can simplify the code by using the `snakeToCamel`.

File: src/lib/types/validator.ts [hunk 3]
```diff
@@ -15,3 +16,50 @@ export const zValidator = z
...
+    rank_cummulative_voting_power: z.string().nullable(),
+    validator_address: zValidatorAddr,
+    account_address: zBechAddr20,
+    identity: z.string(),
+    moniker: z.string(),
+    details: z.string(),
+    commission_rate: z.coerce.number(),
+    is_jailed: z.boolean(),
+    is_active: z.boolean(),
+    voting_power: z.string(),
```
Comment (evilpeach) [inline src/lib/types/validator.ts:32]: Let's use zBig as well.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/types/validator.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import { z } from "zod";
 
-import { zValidatorAddr } from "./addrs";
+import { snakeToCamel } from "lib/utils/formatter/snakeToCamel";
```
Comment (evilpeach) [inline src/lib/types/validator.ts:3]: Let's import from `lib/utls`

File: src/lib/types/validator.ts [hunk 5]
```diff
@@ -15,3 +18,22 @@ export const zValidator = z
...
+    validator_address: zValidatorAddr,
+    account_address: zBechAddr20,
+    identity: z.string(),
+    moniker: z.string(),
+    details: z.string(),
+    commission_rate: z.coerce.number(),
+    is_jailed: z.boolean(),
+    is_active: z.boolean(),
+    voting_power: zBig,
+    uptime: z.number().optional(),
```
Comment (evilpeach) [inline src/lib/types/validator.ts:35]: From my undestanding, uptime can either be a number or zero. Therefore, let's eliminate the `optional` method.

File: src/lib/amplitude/track-event/trackInteraction.ts [hunk 6]
```diff
@@ -1,9 +1,7 @@
-import big from "big.js";
-
 import { amp } from "../Amplitude";
 import { AmpEvent } from "../types";
 import type { SearchResultType } from "lib/services/searchService";
-import type { Addr, Option, Token } from "lib/types";
+import { big, type Addr, type Option, type Token } from "lib/types";
```
Comment (evilpeach) [inline src/lib/amplitude/track-event/trackInteraction.ts:4]: Let's separate into 2 lines like this
```ts
import { big } from "lib/types";
import type { Addr, Option, Token } from "lib/types";
```

File: src/lib/components/delegations/radio-card/MultiBondsRadioCard.tsx [hunk 7]
```diff
@@ -1,8 +1,7 @@
 import { Flex, Text, Heading, Spinner } from "@chakra-ui/react";
 import type { Big } from "big.js";
-import big from "big.js";
 
-import type { Option, TokenWithValue, USD } from "lib/types";
+import { big, type Option, type TokenWithValue, type USD } from "lib/types";
```
Comment (evilpeach) [inline src/lib/components/delegations/radio-card/MultiBondsRadioCard.tsx:4]: ditto

File: src/lib/components/delegations/total-card/MultiBondsCard.tsx [hunk 8]
```diff
@@ -1,10 +1,15 @@
 import { Flex, Heading, Text, useDisclosure } from "@chakra-ui/react";
 import type { Big } from "big.js";
-import big from "big.js";
 import plur from "plur";
 
 import { CustomIcon } from "lib/components/icon";
-import type { BechAddr, Option, TokenWithValue, USD } from "lib/types";
+import {
```
Comment (evilpeach) [inline src/lib/components/delegations/total-card/MultiBondsCard.tsx:6]: ditto

File: src/lib/types/validator.ts (hunk 4 above)
Comment (songwongtp) [reply src/lib/types/validator.ts:3]: cannot import like that, getting cyclic warning

File: src/lib/types/validator.ts (hunk 5 above)
Comment (songwongtp) [reply src/lib/types/validator.ts:35]: optional as it is used for validator info too

File: src/lib/components/delegations/total-card/TotalCardModal.tsx [hunk 9]
```diff
@@ -12,12 +12,11 @@ import {
   Text,
 } from "@chakra-ui/react";
 import type { Big } from "big.js";
-import big from "big.js";
 
 import { CopyLink } from "lib/components/CopyLink";
 import { CustomIcon } from "lib/components/icon";
 import { TokenImageRender, TokenComposition } from "lib/components/token";
-import type { BechAddr, TokenWithValue, USD } from "lib/types";
+import { big, type BechAddr, type TokenWithValue, type USD } from "lib/types";
```
Comment (evilpeach) [inline src/lib/components/delegations/total-card/TotalCardModal.tsx:19]: fix it

File: src/lib/model/account.ts [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
-import type {
-  Option,
-  BechAddr,
-  USD,
-  TokenWithValue,
-  StakingParams,
-  Delegation,
-  Unbonding,
-  Redelegation,
+import {
```
Comment (evilpeach) [inline src/lib/model/account.ts:8]: this

File: src/lib/pages/pools/components/pool-details/tables/pool-txs/messages/lp/components/PoolAssetsGrid.tsx [hunk 11]
```diff
@@ -1,11 +1,10 @@
 import { Flex, SimpleGrid, Text } from "@chakra-ui/react";
 import type { Coin } from "@cosmjs/stargate";
-import big from "big.js";
 
 import { AssetCard, ErrorFetchingDetail } from "../../components";
 import { Loading } from "lib/components/Loading";
 import { useTxData } from "lib/services/txService";
-import type { AssetInfos, Option } from "lib/types";
+import { big, type AssetInfos, type Option } from "lib/types";
```
Comment (evilpeach) [inline src/lib/pages/pools/components/pool-details/tables/pool-txs/messages/lp/components/PoolAssetsGrid.tsx:7]: ditto

File: src/lib/components/upload/UploadCard.tsx [hunk 12]
```diff
@@ -1,9 +1,8 @@
 import { Button, Flex, Text } from "@chakra-ui/react";
-import big from "big.js";
 
 import { AmpEvent, track } from "lib/amplitude";
 import { CustomIcon, UploadIcon } from "lib/components/icon";
-import type { Nullable, Option } from "lib/types";
+import { big, type Nullable, type Option } from "lib/types";
```
Comment (evilpeach) [inline src/lib/components/upload/UploadCard.tsx:5]: ditto

**Review by evilpeach — APPROVED** [review]

**Pull request merged**